service/history/workflow/workflow_task_state_machine.go

```diff
@@ -106,24 +106,26 @@ func (m *workflowTaskStateMachine) ReplicateWorkflowTaskScheduledEvent(
 }
 
 func (m *workflowTaskStateMachine) ReplicateTransientWorkflowTaskScheduled() (*WorkflowTaskInfo, error) {
-	if m.HasPendingWorkflowTask() || m.ms.GetExecutionInfo().WorkflowTaskAttempt == 1 {
+	if !m.ms.IsTransientWorkflowTask() {
```

Comment:
nit: maybe add a comment says caller guarantees there's no pending workflow task when calling this method? (only called after ReplicateWorkflowTaskFailed/TimeoutEvent)

Reply:
I decided to bring `HasPendingWorkflowTask()` check back. This
```go
if m.HasPendingWorkflowTask() || !m.ms.IsTransientWorkflowTask() {
```
looks little bit wired (WT is not there and WT is transient at the same time), but  `IsTransientWorkflowTask` indicates that _next_ WT should be transient, not the current one. Added comment for now, probably need to refactor this even further.